go/cmd/dolt/commands/diffsplitter.go

```diff
@@ -117,7 +118,8 @@ func (ds diffSplitter) splitDiffResultRow(row sql.Row) (rowDiff, rowDiff, error)
 			oldRow.row[ds.queryToTarget[i]] = row[i]
 
 			if diffTypeStr == "modified" {
-				if row[i] != row[ds.fromTo[i]] {
+				// need this to compare map[string]interface{} and other incomparable result types
+				if !reflect.DeepEqual(row[i], row[ds.fromTo[i]]) {
```

Comment:
Probably want to use sql Type.Compare with the type of the result schema if you're going this route